extension/vector/src/function/query_hnsw_index.cpp

```diff
@@ -119,18 +119,20 @@ static std::unique_ptr<TableFuncBindData> bindFunc(main::ClientContext* context,
         outputNode);
 }
 
-static std::vector<float> convertQueryVector(const Value& value) {
-    std::vector<float> queryVector;
+template<typename T>
```

Comment:
can you declare vector index supported types as a concept?
E.g. 
```
template<typename T>
concept vss_element_type = floatingPointTypes
```
This can expedite the  clang-tidy analyzer and make it clear what types are supported in vector index.

Reply:
yeah. good point!